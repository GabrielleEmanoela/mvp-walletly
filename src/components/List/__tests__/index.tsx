import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { ListComponent, ListProps } from '..';
import '@testing-library/jest-native/extend-expect';

const mockProps: ListProps = {
  data: [
    {
      id: '1',
      description: 'Description for transaction 1',
      icon: 'search',
      iconColor: '#000',
      date: '',
      type: 'deposit',
      amount: 0,
      currency: '',
      status: 'completed',
    },
    {
      id: '2',
      description: 'Description for transaction 2',
      icon: 'search',
      iconColor: '#000',
      date: '',
      type: 'deposit',
      amount: 0,
      currency: '',
      status: 'completed',
    },
  ],
  onPress: jest.fn(),
};

const componetName = 'ListComponent';

describe(`${componetName}`, () => {
  it('should call onPress when an item is pressed', async () => {
    const { findByText } = render(<ListComponent {...mockProps} />);
    const itemTitle = await findByText(mockProps.data[0].description);
    fireEvent.press(itemTitle);
    expect(mockProps.onPress).toHaveBeenCalledWith({ id: '1' });
  });
  it('should render empty state when no data is provided', async () => {
    const { findByText } = render(<ListComponent data={[]} />);
    expect(await findByText('Nenhuma transação encontrada.')).toBeTruthy();
  });

  it('should render correctly with testID', () => {
    const { getByTestId } = render(<ListComponent {...mockProps} />);
    expect(getByTestId('listComponent')).toBeTruthy();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<ListComponent {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
