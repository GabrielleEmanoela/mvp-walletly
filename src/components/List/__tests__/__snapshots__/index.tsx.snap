// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID="listComponent"
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    contentContainerStyle={
      {
        "backgroundColor": "#fff",
        "paddingVertical": 8,
      }
    }
    data={
      [
        {
          "amount": 0,
          "currency": "",
          "date": "",
          "description": "Description for transaction 1",
          "icon": "search",
          "iconColor": "#000",
          "id": "1",
          "status": "completed",
          "type": "deposit",
        },
        {
          "amount": 0,
          "currency": "",
          "date": "",
          "description": "Description for transaction 2",
          "icon": "search",
          "iconColor": "#000",
          "id": "2",
          "status": "completed",
          "type": "deposit",
        },
      ]
    }
    estimatedItemSize={80}
    getItem={[Function]}
    getItemCount={[Function]}
    initialScrollIndex={0}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          testID="item-1"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                [
                  {
                    "paddingRight": 24,
                    "paddingVertical": 8,
                  },
                  {
                    "backgroundColor": "#fff",
                    "borderRadius": 8,
                    "marginVertical": 4,
                    "paddingHorizontal": 12,
                    "paddingVertical": 8,
                  },
                ],
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "marginVertical": 6,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                color="#000"
                style={
                  [
                    {
                      "backgroundColor": "#4d539bff",
                      "borderRadius": 20,
                      "height": 40,
                      "width": 40,
                    },
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "color": "#000",
                        "fontSize": 24,
                      },
                    ]
                  }
                >
                  â–¡
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "paddingLeft": 16,
                    },
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                testID="undefined-content"
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Description for transaction 1
                </Text>
              </View>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    {
                      "alignSelf": "center",
                      "color": "#333",
                      "flexWrap": "wrap",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "maxWidth": 120,
                    },
                  ]
                }
              >
                 
                0
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#f0f0f0",
              "height": 1,
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          testID="item-2"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                [
                  {
                    "paddingRight": 24,
                    "paddingVertical": 8,
                  },
                  {
                    "backgroundColor": "#fff",
                    "borderRadius": 8,
                    "marginVertical": 4,
                    "paddingHorizontal": 12,
                    "paddingVertical": 8,
                  },
                ],
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "marginVertical": 6,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                color="#000"
                style={
                  [
                    {
                      "backgroundColor": "#4d539bff",
                      "borderRadius": 20,
                      "height": 40,
                      "width": 40,
                    },
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "color": "#000",
                        "fontSize": 24,
                      },
                    ]
                  }
                >
                  â–¡
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "paddingLeft": 16,
                    },
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                testID="undefined-content"
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Description for transaction 2
                </Text>
              </View>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    {
                      "alignSelf": "center",
                      "color": "#333",
                      "flexWrap": "wrap",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "maxWidth": 120,
                    },
                  ]
                }
              >
                 
                0
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
